### PR TITLE
fix: ip discovery changes (74 byte udp packets)

### LINKED ISSUE #1954: IP Discovery
### Summary

Discord recently pushed out a change related to Voice Connections that broke apps sending 70-byte UDP packets to the voice server when using IP Discovery.

### What is the feature request for?

The core library

### The Problem

Starting March 15, 2023, all apps with Voice Connections using IP Discovery must send 74-byte UDP packets. IP Discovery requests sending the deprecated 70-byte packet will no longer receive a response.
﻿
https://discord.com/developers/docs/topics/voice-connections#ip-discovery

### The Ideal Solution

Sending 74-byte UDP packets.

### The Current Solution

N/A

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Lulalaby**
Closes #1954

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ These changes are available on the `master` branch, but have not yet been releas
 
 - Fixed the type-hinting of `SlashCommandGroup.walk_commands()` to reflect actual
   behavior. ([#1852](https://github.com/Pycord-Development/pycord/pull/1852))
+- Fixed the voice ip discovery due to recent [announced change](https://discord.com/channels/613425648685547541/697138785317814292/1080623873629884486)
```
Comment (Lulalaby) [inline CHANGELOG.md]:
```suggestion
- Fixed the voice ip discovery due to the recent [announced change](https://discord.com/channels/613425648685547541/697138785317814292/1080623873629884486) 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,9 @@ These changes are available on the `master` branch, but have not yet been releas
 
 - Fixed the type-hinting of `SlashCommandGroup.walk_commands()` to reflect actual
   behavior. ([#1852](https://github.com/Pycord-Development/pycord/pull/1852))
+- Fixed the voice ip discovery due to recent
```
Comment (Lulalaby) [inline CHANGELOG.md:27]:
```suggestion
- Fixed the voice ip discovery due to the recent
```

File: discord/gateway.py [hunk 3]
```diff
@@ -892,16 +892,16 @@ async def initial_connection(self, data):
         state.voice_port = data["port"]
         state.endpoint_ip = data["ip"]
 
-        packet = bytearray(70)
+        packet = bytearray(74)
         struct.pack_into(">H", packet, 0, 1)  # 1 = Send
         struct.pack_into(">H", packet, 2, 70)  # 70 = Length
         struct.pack_into(">I", packet, 4, state.ssrc)
```
Comment (Lulalaby) [inline discord/gateway.py:898]: The fixes were made by minn on dpy

I'm actually not sure if that should change too.
They did not changed it there, but the 70 confuses me.

I don't have a clue what i'm doing here.

@BobDotCom do you have an idea?

File: CHANGELOG.md [hunk 4]
```diff
@@ -24,6 +24,9 @@ These changes are available on the `master` branch, but have not yet been releas
 
 - Fixed the type-hinting of `SlashCommandGroup.walk_commands()` to reflect actual
   behavior. ([#1852](https://github.com/Pycord-Development/pycord/pull/1852))
+- Fixed the voice ip discovery due to the recent
+  [announced change](https://discord.com/channels/613425648685547541/697138785317814292/1080623873629884486)
```
Comment (BobDotCom) [inline CHANGELOG.md:28]:
```suggestion
  [announced change](https://discord.com/channels/613425648685547541/697138785317814292/1080623873629884486).
```

**Review by JustaSqu1d — APPROVED** [review]

**Review by NeloBlivion — APPROVED** [review]
LGTM and tested fine, though this technically isn't a new change; it's been 74 bytes [since December 2019](https://github.com/discord/discord-api-docs/commit/f90a59d1d529ac7e9737fc770218e05efe0d7e3c) while 70 bytes has been deprecated since then, and now they're finally removing it completely. 
This could be made clearer in the changelog, but the difference isn't particularly important.

**Pull request merged**